src/components/main-view/main-view.jsx

```diff
@@ -1,25 +1,217 @@
-import React from 'react';
+import { useState, useEffect } from 'react';
+
+// Importing MovieCard component, so it can be used here
 import { MovieCard } from '../movie-card/movie-card';
 
-export class MainView extends React.Component {
+// Importing MovieView component
+import { MovieView } from '../movie-view/movie-view';
+
+import { LoginView } from '../login-view/login-view';
+import { SignupView } from '../signup-view/signup-view';
+import { NavigationBar } from '../navigation-bar/navigation-bar';
+
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+import Button from 'react-bootstrap/Button';
 
-  constructor(){
-    super();
+import { BrowserRouter, Routes, Route, Navigate } from 'react-router-dom';
+import Container from 'react-bootstrap/Container';
+import { ProfileView } from '../profile-view/profile-view';
 
-    this.state = {
-      movies: [{title: "Clash of Titans"}, {title: "Guardians of the Galaxy"}]
+export const MainView = () => {
+  const storedUser = JSON.parse(localStorage.getItem('user'));
+  const storedToken = localStorage.getItem('token');
+  const [movies, setMovies] = useState([]);
+  const [selectedMovie, setSelectedMovie] = useState(null);
+  const [user, setUser] = useState(storedUser ? storedUser : null);
+  const [token, setToken] = useState(storedToken ? storedToken : null);
+
+  useEffect(() => {
+    if (!token) {
+      return;
     }
-  }
-
-  render() {
-    const { movies } = this.state;
-  
-    if (movies.length === 0) return <div className="main-view">The list is empty!</div>;
-  
-    return (
-      <div className="main-view">
-        {movies.map((movieItem, index) => <MovieCard movie={movieItem} key={index} />)}
-      </div>
-    );
-  }
-}
+    
+
+
+    fetch('https://myflixapi-50jx.onrender.com/movies',  {
```

Comment:
its made by khaled i dont know if is the corect to do the old url was importtin from heroko